pkg/storage/batch_test.go

```diff
@@ -1013,3 +1013,33 @@ func TestBatchReader(t *testing.T) {
 	require.False(t, r.Next())
 	require.NoError(t, r.Error())
 }
+
+func TestBatchReaderVerify(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	DisableMetamorphicSimpleValueEncoding(t)
+
+	eng := NewDefaultInMemForTesting()
+	defer eng.Close()
+
+	b := eng.NewBatch()
+	defer b.Close()
+
+	// Write some basic data.
+	require.NoError(t, b.PutMVCC(pointKey("mvccKey", 1), stringValue("mvccValue")))
+	r, err := NewBatchReader(b.Repr())
+	require.NoError(t, err)
+	require.True(t, r.Next())
+	mvccKey, _ := r.MVCCKey()
+	// Initalize checksum in mvcc value
+	mvccValue, _, _ := tryDecodeSimpleMVCCValue(r.Value())
+	mvccValue.Value.InitChecksum(mvccKey.Key)
+	err = r.Verify()
+	require.NoError(t, err)
+	// Simulate data corruption
+	r.value[len(r.value)-1]++
+	err = r.Verify()
+	// Throw err on checksum mismatch
+	require.NotNil(t, err)
```

Comment:
```suggestion
	// Throw err on checksum mismatch
	require.NotNil(t, r.Verify())
```

Also, consider using `require.ErrorContains`, to check the error message.